sklearn/impute/_base.py

```diff
@@ -113,6 +114,14 @@ def _concatenate_indicator(self, X_imputed, X_indicator):
 
         return hstack((X_imputed, X_indicator))
 
+    def _concatenate_indicator_feature_names_out(self, names, input_features):
+        if not self.add_indicator:
+            return names
+
+        indicator_names = self.indicator_.get_feature_names_out(input_features)
+        indicator_names_with_prefix = [f"indicator_{name}" for name in indicator_names]
```

Comment:
I find the `indicator_` prefix not explicit enough. I would rather suggest `missing_` or `imputed_` to be specific that this is a missingness or imputation indicator column.

We could use `missing_indicator_` to be extra explicit but I have the feeling this would be necessary verybose.

Reply:
If we follow the `SLEP007`, then we use `missingindicator_` as the prefix since it is the class name for the output of `MissingIndicator.get_feature_names_out`.

I updated the PR to reflect this, but I am open to changing it.